src/ray/util/tests/pipe_logger_test.cc

```diff
@@ -178,4 +202,35 @@ TEST(PipeLoggerTestWithTee, RotatedRedirectionWithTee) {
 
 }  // namespace ray
 
+#elif defined(_WIN32)
+
+#include <windows.h>
+
+#include "ray/util/tests/windows_test_utils.h"
+
+namespace ray {
+
+TEST(PipeLoggerTestWithTee, RedirectionWithNoTeeAndRotation) {
```

Comment:
why do we need to write the same test twice? I think you can write one that's runnable in both linux and windows

Reply:
We could do that, but we need a unified write interface for both platforms first.

Reply:
I see your write code is exactly the same in these two places?

Reply:
At minimum, we should have #ifdef inside the test, wrap the platform specific code so we don't need to duplicate remaining code.

Reply:
Should be good now, I use write function in the handle, and STL filesystem APIs.